local_ads/statistics.cpp

```diff
@@ -381,21 +384,40 @@ void Statistics::ProcessEvents(std::list<Event> & events)
 
 void Statistics::SendToServer()
 {
+  std::string userId;
+  ServerSerializer serializer;
+  {
+    std::lock_guard<std::mutex> lock(m_mutex);
+    userId = m_userId;
+    serializer = m_serverSerializer;
+  }
+
   for (auto it = m_metadataCache.begin(); it != m_metadataCache.end();)
   {
     std::string const url = MakeRemoteURL(m_userId, it->first.first, it->first.second);
     if (url.empty())
       return;
 
-    std::vector<uint8_t> bytes = SerializeForServer(ReadEvents(it->second.m_fileName));
-    if (bytes.empty())
+    std::list<Event> events = ReadEvents(it->second.m_fileName);
+    if (events.empty())
     {
       ++it;
       continue;
     }
 
+    std::string contentType = "application/octet-stream";
+    std::string contentEncoding = "";
+    std::vector<uint8_t> bytes = serializer != nullptr
+                                     ? serializer(events, userId, contentType, contentEncoding)
+                                     : SerializeForServer(events);
+    ASSERT(!bytes.empty(), ());
+
     platform::HttpClient request(url);
-    request.SetBodyData(std::string(bytes.begin(), bytes.end()), "application/octet-stream");
+#ifdef DEV_LOCAL_ADS_SERVER
```

Comment:
Why do you need these ugly ifdefs? Why not to create a simple C++ class, say, DevLocalAdsServer, that will take the content, prepare headers and run http request?